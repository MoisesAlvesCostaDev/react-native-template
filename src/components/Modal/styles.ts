import styled from "styled-components/native";
import { BACKGROUND_COLOR_PRIMARY,TITLE_SCREEN_TEXT_COLOR,BACKGROUND_COLOR_SECONDARY } from "assets/colors";

export const CenteredView = styled.View`
  margin-top: 50px;
position: absolute;
height: 100%;
width: 100%;
justify-content: center;
align-items: center;
`;

export const ModalView = styled.View`
    background-color: ${BACKGROUND_COLOR_PRIMARY};
    border-radius: 20px;
    padding: 15px;
    align-items: center;
    width: 290px;
    height: 157px;
`;


export const HeaderView = styled.View`
width: 100%;
height: 30px;
justify-content: center;
align-items: center;
`;

export const MessageText = styled.Text`
color: ${TITLE_SCREEN_TEXT_COLOR};
font-size: 16px;
`;


export const BodyView = styled.View`
margin-top:10px
width: 100%;
height: 30px;
align-items: center;
height: 60px;
`;

export const FooterView = styled.View`
width: 100%;
height: 30px;
justify-content: center;
align-items: center;
`;


export const ModalButton = styled.TouchableOpacity`
background-color: ${BACKGROUND_COLOR_SECONDARY};
width: 120px
height: 30px;
justify-content: center;
align-items: center;
border-radius:30px;
`;


export const ButtomText = styled.Text`
font-size: 16px;
font-weight: bold;
`;
