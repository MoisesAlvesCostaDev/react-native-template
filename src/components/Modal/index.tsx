import React  from "react";
import { Modal as ReactModal} from "react-native";
import * as styled from "./styles";
import { TitleScreen } from "assets/styles/Screen";

const Modal = ({
  modalVisible,
  setModalVisible,
}: {
  modalVisible: boolean;
  setModalVisible: React.Dispatch<React.SetStateAction<boolean>>;
}): JSX.Element => {
  return (
    <styled.CenteredView>
      <ReactModal
        animationType="slide"
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}
      >
        <styled.CenteredView>
          <styled.ModalView>
            <styled.HeaderView>
              <TitleScreen>Não foi possível entrar</TitleScreen>
            </styled.HeaderView>
            <styled.BodyView>
              <styled.MessageText>Usuáro ou senha inválidos</styled.MessageText>
            </styled.BodyView>
            <styled.FooterView>
              <styled.ModalButton
              testID="buttonStyled"
                onPress={() => setModalVisible(!modalVisible)}
              >
                <styled.ButtomText>Ok</styled.ButtomText>
              </styled.ModalButton>
            </styled.FooterView>
          </styled.ModalView>
        </styled.CenteredView>
      </ReactModal>
    </styled.CenteredView>
  );
};

export default Modal;
