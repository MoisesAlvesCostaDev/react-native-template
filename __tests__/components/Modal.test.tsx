import "react-native";
import React from "react";
import Modal from "components/Modal";
import { fireEvent, render } from "@testing-library/react-native";


describe("test render component with title and message default", () => {
  it("renders correctly", () => {
    const { getByTestId, getByText, toJSON } = render(
      <Modal modalVisible = {true} setModalVisible={()=>{true}} />
    );
    getByText("Não foi possível entrar");
    getByText("Usuáro ou senha inválidos");
    expect(toJSON()).toMatchSnapshot();
  });
});

describe("test render and click button ok", () => {
  it("renders correctly", () => {
    const { getByTestId } = render( <Modal modalVisible = {true} setModalVisible={()=>{true}} />);
    const buttonStyled = getByTestId("buttonStyled");
    fireEvent.press(buttonStyled);
  });
});
