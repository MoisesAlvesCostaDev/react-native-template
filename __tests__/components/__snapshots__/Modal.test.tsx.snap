// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render component with title and message default renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "marginTop": 50,
        "position": "absolute",
        "width": "100%",
      },
    ]
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "marginTop": 50,
            "position": "absolute",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF5C00",
              "borderRadius": 20,
              "height": 157,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
              "width": 290,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 30,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Não foi possível entrar
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 60,
                "marginTop": 10,
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                },
              ]
            }
          >
            Usuáro ou senha inválidos
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 30,
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 30,
                "height": 30,
                "justifyContent": "center",
                "opacity": 1,
                "width": 120,
              }
            }
            testID="buttonStyled"
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Ok
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
